routes/index.js

```diff
@@ -1,71 +1,86 @@
+router.post('/user/register', function(req, res, next) {
     var username = req.body.user_name;
     var password = req.body.password;
     var cpassword = req.body.confirm_password;
     var email = req.body.email;
     var firstname = req.body.first_name;
-    var lastname = req.body.last_name; 
-    var pass=crypto.createHash('md5').update(password).digest('hex');
-    if((username.length >0) && (password.length >0) && (cpassword.length >0) && (email.length >0)&&(firstname.length >0)&&(lastname.length >0)){
-        if(password == cpassword) {
+    var lastname = req.body.last_name;
+    var pass = crypto.createHash('md5').update(password).digest('hex');
+    if ((username.length > 0) && (password.length > 0) && (cpassword.length > 0) && (email.length > 0) && (firstname.length > 0) && (lastname.length > 0)) {
+        if (password == cpassword) {
             var record = new req.users({
-               "username": username,
-               "password": pass,
-               "email": email,
-               "firstname": firstname,
-               "lastname": lastname,
+                "username": username,
+                "password": pass,
+                "email": email,
+                "firstname": firstname,
+                "lastname": lastname,
             });
-            record.save(function(err,details) {
+            record.save(function(err, details) {
                 if (err) {
-                   res.json("Username or Email already exists");
+                    next("user already exist");
                 } else {
-                   res.json("Record inserted successfully");
+                    next("Record inserted successfully");
                 }
             });
-        }else{
-         res.json("Password is not matched")
-        } 
-    }else{
-        res.json("All field must be filled out");
-    }        
+        } else {
+            next("Password is not matched")
+        }
+    } else {
+        next("All field must be filled out");
+    }
 });
 
 <!--------- login -------->
-router.post('/user/login', function(req, res) {  
+
+router.post('/user/login', function(req, res, next) {
     var username = req.body.user_name;
     var password = req.body.password;
-    var pass=crypto.createHash('md5').update(password).digest('hex');
+    var pass = crypto.createHash('md5').update(password).digest('hex');
     req.users.findOne({
-        "username":username,
+        "username": username,
     }, function(err, docs) {
         if (err) {
             res.json("Your username is not exist");
-        }else if(pass == docs.password){
-          var access_token=docs._id;
-            res.json({'access_token':docs._id});  
-        }else{
-            res.status(500).send({status:0,message:"Invalid password"});
-        }  
+        } else if (pass == docs.password) {
+            var now = moment().format("YYYY-MM-DD'T'HH:mm:ss:SSSZ"); // save date in proper format....
```

Comment:
token should be a random number. not current time. 
make token md5(time()) and save expiry_at seperately in collection